docs/react-native/learn/todo-app/app-hands-on/navigation-hook.mdx

```diff
@@ -0,0 +1,133 @@
+---
+title: ナビゲーションの型定義
+---
+
+## useNavigationのカスタムフック
+
+[useNavigationの型定義](../../basic-concepts/react-navigation-basics/modal#usenavigationの型定義)にあるとおり、`useNavigation`に型を指定することで、型安全にナビゲーション操作できます。
```

Comment:
https://reactnavigation.org/docs/typescript/#annotating-usenavigation では、

> To annotate the navigation prop that we get from useNavigation, we can use a type parameter:
>
> const navigation = useNavigation&lt;ProfileScreenNavigationProp&gt;();
>
> It's important to note that this isn't completely type-safe because the type parameter you use may not be correct and we cannot statically verify it.

とあえて補足されているので、"正しい"型を指定することでと条件付きにしておきたいです。

また、カスタムフックを用意する理由の説明が不足しているように感じたので、
"正しい"型を指定すれば型安全だけど、複雑なので正しい型の指定をメンテナンスし続けるのは大変(かつ間違ってても気づきにくい)なので、管理を楽にするためにカスタムフックを用意するという話の流れにしても良いかと思いました。

Reply:
確かに理由が不足していますね。上記文章を元に理由を追記しました。

https://github.com/ws-4020/mobile-app-crib-notes/pull/55/commits/6c670f5b183466c0773565742141a68eb6e3cd68